.circleci/config.yml

```diff
@@ -2,53 +2,100 @@ version: 2
 # Tell CircleCI to use this workflow
 workflows:
   version: 2
+  nightly:
+    triggers:
+      - schedule:
+          cron: "0 0 * * *"
+          filters:
+            branches:
+              only:
+                - master
+    jobs:
+      - "upstream-dev"
+      - "python-3.6"
+      - "python-3.7"
+      - "python-3.8"
   default:
     jobs:
-      - "build"
-
-jobs:
-  "build":
-    docker:
-      - image: continuumio/miniconda3:latest
-
-    parallelism: 5
-
-    environment:
-      ENV_NAME: "notebook-gallery"
-      TEST_FILES: "notebooks/Run-Anywhere/**/*.ipynb"
+      - "upstream-dev"
+      - "python-3.6"
+      - "python-3.7"
+      - "python-3.8"
 
+shared: &shared
     steps:
-
-      # Get our data and merge with upstream
       - checkout
-      # Restore cached files to speed things up
+      - run:
+          name: Environment Checksum
+          command: |
+            # Get around limitation of using custom ENV variables in checksum key
+            cp ${ENV_FILE} _env_copy
+
       - restore_cache:
-          key: deps-{{ .Environment.CIRCLE_PROJECT_REPONAME }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "environment.yml" }}
-      - run: # install and activate conda environment
+          key: deps-{{ .Environment.CIRCLE_PROJECT_REPONAME }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "_env_copy" }}
+
+      - run:
           name: Install conda environment
           command: |
-            conda env update -f environment.yml -q
+            conda env update -f ${ENV_FILE} -q
```

Comment:
If you look closely at intake-esm's CI config, we don't create a new environment. We update the base environment `conda env update -n base -f ${ENV_FILE}`.

Reply:
Ah, thank you.   This could explain why the python version is changing for some builds.

Reply:
Do you want me to modify the intake-esm CI config to calculate checksums based on the environment file being used?